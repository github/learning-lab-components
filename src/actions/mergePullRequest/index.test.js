const mergePullRequest = require('.')
const mockContext = require('../../tests/mockContext')

describe('mergePullRequest', () => {
  let context

  beforeEach(() => {
    context = mockContext({
      pull_request: {
        number: 1
      }
    }, {
      pullRequests: {
        merge: jest.fn()
      },
      search: {
        issues: jest.fn(() => Promise.resolve({
          data: {
            items: [{ title: 'My Pull Request', number: 1 }]
          }
        }))
      }
    })
  })

  it('merges a pull request', async () => {
    await mergePullRequest(context, {})
    expect(context.github.pullRequests.merge).toHaveBeenCalled()
    expect(context.github.pullRequests.merge.mock.calls).toMatchSnapshot()
  })

  it('merges a pull request with a title string', async () => {
    await mergePullRequest(context, { pullRequest: 'My Pull Request' })
    expect(context.github.pullRequests.merge).toHaveBeenCalled()
    expect(context.github.pullRequests.merge.mock.calls).toMatchSnapshot()
  })

  it('merges a pull request with a pr number', async () => {
    await mergePullRequest(context, { pullRequest: 1 })
    expect(context.github.pullRequests.merge).toHaveBeenCalled()
    expect(context.github.pullRequests.merge.mock.calls).toMatchSnapshot()
  })
})
