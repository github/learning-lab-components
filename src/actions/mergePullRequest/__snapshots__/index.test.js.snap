// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mergePullRequest merges a pull request 1`] = `
Array [
  Array [
    Object {
      "number": 1,
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;

exports[`mergePullRequest merges a pull request with a pr number 1`] = `
Array [
  Array [
    Object {
      "number": 1,
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;

exports[`mergePullRequest merges a pull request with a title string 1`] = `
Array [
  Array [
    Object {
      "number": 1,
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;
